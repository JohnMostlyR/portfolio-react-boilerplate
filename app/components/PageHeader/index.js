import React from 'react';
import PropTypes from 'prop-types';

import StyledHeader from './StyledHeader';
import H2 from '../H2';
import SpeechBubble from '../SpeechBubble';
import { theme } from '../../styles/theme';

const {
  pageHeader: { backgroundColor, color },
} = theme;

const PageHeader = ({ arrowPosition, children }) => (
  <StyledHeader>
    <SpeechBubble
      arrowHeight="4vh"
      arrowPosition={arrowPosition}
      backgroundColor={backgroundColor}
      makeAppear
      showArrowBreakpoint="750px"
    >
      <H2 color={color}>{children}</H2>
    </SpeechBubble>
  </StyledHeader>
);

PageHeader.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  arrowPosition: PropTypes.oneOf([
    'bottom-left',
    'bottom-right',
    'top-left',
    'top-right',
  ]),
};

export default PageHeader;
