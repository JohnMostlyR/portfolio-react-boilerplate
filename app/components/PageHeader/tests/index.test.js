import React from 'react';
import { shallow, mount } from 'enzyme';

import PageHeader from '../index';

describe('<PageHeader />', () => {
  it('should render and match the snapshot', () => {
    const wrapper = shallow(<PageHeader />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should adopt the "isLeftHanded" prop', () => {
    const isLeftHanded = false;
    const wrapper = mount(<PageHeader isLeftHanded={isLeftHanded} />);
    expect(wrapper.prop('isLeftHanded')).toBe(isLeftHanded);
  });

  it('should adopt any children', () => {
    const wrapper = shallow(
      <PageHeader>
        <div>children</div>
      </PageHeader>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
