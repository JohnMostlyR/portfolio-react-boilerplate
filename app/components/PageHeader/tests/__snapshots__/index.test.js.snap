// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageHeader /> should adopt any children 1`] = `
<StyledHeader>
  <SpeechBubble
    arrowHeight="4vh"
    arrowPosition="bottom-left"
    backgroundColor="#465a65"
    isGhost={false}
    makeAppear={true}
    showArrowBreakpoint="750px"
  >
    <H2
      color="#aedd2c"
    >
      <div>
        children
      </div>
    </H2>
  </SpeechBubble>
</StyledHeader>
`;

exports[`<PageHeader /> should render and match the snapshot 1`] = `
<StyledHeader>
  <SpeechBubble
    arrowHeight="4vh"
    arrowPosition="bottom-left"
    backgroundColor="#465a65"
    isGhost={false}
    makeAppear={true}
    showArrowBreakpoint="750px"
  >
    <H2
      color="#aedd2c"
    />
  </SpeechBubble>
</StyledHeader>
`;
